tests/query/test_query_transaction.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from ydb.query.transaction import BaseTxContext
+from ydb.query.transaction import QueryTxStateEnum
+
+
+class TestQueryTransaction:
+    def test_tx_begin(self, tx: BaseTxContext):
+        assert tx.tx_id is None
+
+        tx.begin()
+        assert tx.tx_id is not None
+
+    def test_tx_allow_double_commit(self, tx: BaseTxContext):
+        tx.begin()
+        tx.commit()
+        tx.commit()
+
+    def test_tx_allow_double_rollback(self, tx: BaseTxContext):
+        tx.begin()
+        tx.rollback()
+        tx.rollback()
+
+    def test_tx_commit_raises_before_begin(self, tx: BaseTxContext):
+        with pytest.raises(RuntimeError):
+            tx.commit()
```

Comment:
if transaction can implicit begin - it should allow to commit/rollback without begin too.

for prevent unexpected error in case:
```python

if a:
    tx.execute("UPSERT INTO a VALUES(1)")
if b:
    tx.execute("UPSERT INTO b VALUES(2)")

tx.commit()
```

in the case if one of a or b will be true = code is valid, but if a and b will be false - it cause commit without any previous action for the transaction and now it will raise exeption. But should be work fine.